source/common/event/file_event_impl.cc

```diff
@@ -31,6 +31,11 @@ FileEventImpl::FileEventImpl(DispatcherImpl& dispatcher, os_fd_t fd, FileReadyCb
   RELEASE_ASSERT(trigger_ != FileTriggerType::Edge,
                  "libevent does not support edge triggers on Windows");
 #endif
+  if constexpr (PlatformDefaultTriggerType != FileTriggerType::EmulatedEdge) {
+    RELEASE_ASSERT(trigger_ != FileTriggerType::EmulatedEdge,
```

Comment:
nit: Change to ASSERT for performance.

Reply:
fixed here and above 342fe7383e53537121d9003b2731bda5640f2da8